### PR TITLE
Add 'instance class fields / resolving identifier in parent scope'

### PR DESCRIPTION
**Pull request by davidtaylorhq**
This is to detect https://bugs.webkit.org/show_bug.cgi?id=236843, which affects Safari 15 and below.

Discussion in https://github.com/babel/babel/issues/14289

### CODE REVIEW

File: data-es2016plus.js [hunk 1]
```diff
@@ -5776,6 +5776,39 @@ exports.tests = [
...
           rhino1_7_13: false
         }
+      },
+      {
+        name: 'resolving identifier in parent scope',
+        exec: function () {/*
+          {
+            let a = ["hello world"];
+            class MyClass {
+              c = a[(0)];
```
Comment (ljharb) [inline data-es2016plus.js:5787]: are the parens needed here to trigger the issue?

File: data-es2016plus.js [hunk 2]
```diff
@@ -5776,6 +5776,39 @@ exports.tests = [
...
+          safari15: false,
+          safari16: true,
+          safaritp: true,
+          duktape2_0: false,
+          graalvm19: false,
+          graalvm20: true,
+          hermes0_7_0: false,
+          reactnative0_70_3: true,
+          rhino1_7_13: false
+        }
```
Comment (davidtaylorhq) [inline data-es2016plus.js]: Here, I have copy/pasted the data from 'instance class fields / public instance class fields', and then manually modified the entries for safari. Is that the correct approach? Or is there some automated system to collect this data?

File: data-es2016plus.js (hunk 1 above)
Comment (davidtaylorhq) [reply data-es2016plus.js:5787]: Yes, the parentheses are important. The JIT bug only occurs when parenthesis are included in the initializer.

Comment (ljharb) [reply data-es2016plus.js:5787]: A comment to that affect on this line would be good, to ensure nobody removes them in the future.

File: data-es2016plus.js (hunk 2 above)
Comment (ljharb) [reply data-es2016plus.js]: Ideally, results should only be present in the file if a human has manually verified them.

File: data-es2016plus.js (hunk 1 above)
Comment (davidtaylorhq) [reply data-es2016plus.js:5787]: Added :+1:

File: data-es2016plus.js (hunk 2 above)
Comment (davidtaylorhq) [reply data-es2016plus.js]: Ok that makes sense. Updated to only include browsers I have manually tested

**Comment by nicolo-ribaudo** [comment]
I can confirm that this passes in Firefox 125 and Chrome 74.

**Comment by nicolo-ribaudo** [comment]
Also Firefox 69

**Review by chicoxyzzy — APPROVED** [review]

**Review by ljharb — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
Note: Firefox 69 and Chrome 74 are _not_ included in this PR yet! @davidtaylorhq do you mind opening a new one? :)

**Comment by davidtaylorhq** [comment]
Thanks for the merge! Added those two extra data points in https://github.com/compat-table/compat-table/pull/1899

**Pull request merged**